agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -49,6 +65,36 @@ public static boolean logSpecialOneTimeFriendlyException(
       //  e.g. wrong host address or cannot reach address due to network issues...
       return false;
     }
+
+    IOException ioException = getCausedByOfType(error, IOException.class);
+    if (ioException != null) {
+      boolean foundCiphers = true;
+      try {
+        foundCiphers = hasExpectedCiphers();
+      } catch (NoSuchAlgorithmException e) {
+        logger.debug(e.getMessage(), e);
+        return false;
+      }
+      if (!foundCiphers) {
+        if (!alreadySeen.getAndSet(true)) {
+          logger.error(getCipherFriendlyMessage(url));
+        }
+      } else {
+        return false;
+      }
+      return true;
+    }
```

Comment:
can we add tests for this? there are different code paths here.  not sure if it's easy.

line 82 else block is not necessary, but without it, why line 85 return true again? 
once the error has been handled, can we return true after line 80?